### ISSUE TITLE
OccTax : Filtre Taxons

### ISSUE THREAD

**Issue by sig-pnrnm** [PnX-SI/GeoNature#392]
Bonjour à tous,

Un suggestion relative au Filtre Taxon du formulaire OccTax.
Actuellement (mais c'est peut-être provisoire), il est basé sur 2 listes déroulantes successives : `regne` puis `group2_inpn` de TaxRef.
Personnellement, je ne trouve pas ça très ergonomique. En plus, les règnes ont leur dénomination scientifique latine (Animalia, Plantae, ...).
![filtre_taxon_occtax](https://user-images.githubusercontent.com/26482054/43002884-b481be38-8c2a-11e8-8810-8313660c9795.png)

Certe, l'outil vise un public scientifique, mais je suggèrerais (en remplacement ou en complément) une sélection plus visuelle des groupes taxonomiques naturalistes.
J'ai fait un essai sur la base des pictos présents dans GeoNature-Atlas, et voici ce que ça donnerait : 
![filtre_taxon](https://user-images.githubusercontent.com/26482054/43002960-fd61e6c8-8c2a-11e8-9ea9-e48501ec510f.png)
([la même chose de manière interactive](https://jsfiddle.net/u4cya29p/))

Il y a quelques "arrangements" taxonomiques (ou pourrait par exemple renommer "Plantes & Co"), mais ça ne me semble pas dramatique.
Pour ce qui est du groupe "Insectes", mais c'est certainement une déformation personnelle en tant qu'entomologiste, on pourrait imaginer des sous-groupes pour les principales disciplines (généralement, les ordres, mais parfois plus larges)

Désolé si tout cela était déjà prévu, ou si cela a été exclu pour des raisons que je n'ai pas, mais c'était ma petite suggestion du jour.

Bon courage pour vos développements estivaux !

Sylvain

**Comment by camillemonchicourt** [comment]
Oui pourquoi pas. A creuser. 
A noter cependant que si tu choisis un règne, ça filtre la liste des groupes. Et il faut que l'on trie les règnes par ordre alphabétique.

**Comment by sig-pnrnm** [comment]
> A noter cependant que si tu choisis un règne, ça filtre la liste des groupes.

Oui, j'ai bien vu : mais sur la capture d'écran, tu as bien uniquement les groupes du règle `Animalia`.
D'où le manque d'ergonomie, avec de nombreux groupes qui pourraient être masqués par défaut et associés dans la boite "Autres".

Après, comme je le dis, l'un n'empêche pas l'autre : on peut avoir une sélection par listes ET/OU par boutons (dans mon exemple, je les ai regroupés dans 3 boutons, mais on peut même les regrouper en 1 seul bouton qui se développe en 2 sous-niveaux)

**Comment by DonovanMaillard** [comment]
Les pictos sont plus "sympas" mais je trouve qu'ils sont davantage ambigüs et pas facilement représentatifs des groupes 2 INPN. L'exemple des insectes est toujours facile, ici tous le monde va chercher où saisir une libellule... 

Au choix, sinon il faut pouvoir filtrer sur les ordres, on en a déjà parlé, et je pense que pour notre part on devra faire ces modifications à termes, car le groupe "insectes" pour un pôle invertébrés va regrouper 90% de nos données. Mais la liste des termes, pour ce coup, est moins ambigue, à voir comment concilier les 2

**Comment by camillemonchicourt** [comment]
Par contre un truc que l'on n'a pas fait mais qu'il faudrait faire est de n'afficher que les règnes et groupes correspondant aux espèces présentes dans la liste des taxons. 
Bon dans tous les cas, maintenant que la recherche d'un taxon fonctionne par auto-complétion, ces filtres par règnes et groupes n'ont qu'un intérêt limité.

**Comment by DonovanMaillard** [comment]
Oui et non, à une échelle très large comme nationale, l'autocomplétion peut concerner suffisamment de noms du taxref pour que les filtres restent intéressants :)

**Comment by sig-pnrnm** [comment]
> Les pictos sont plus "sympas" mais je trouve qu'ils sont davantage ambigüs et pas facilement représentatifs des groupes 2 INPN. L'exemple des insectes est toujours facile, ici tout le monde va chercher où saisir une libellule...

C'est bien pour ça que je disais que, pour le groupe "Insectes", il faudrait un accès aux sous-disciplines entomologiques (Odonates, Rhopalocères, Macro-hétérocères, Orthoptères, ...) : 

> on pourrait imaginer des sous-groupes pour les principales disciplines (généralement, les ordres, mais parfois plus larges)


Sinon, pour ce qui est de l'ambiguité des pictos, ceux-ci ne sont pas figés et on pourrait travailler sur des pictos moins ambigus, adaptés à ce contexte. Celui du papillon est en effet très trompeur. Mais il est d'ailleurs également trompeur dans les fiches de Geonature-Atlas !

**Comment by DonovanMaillard** [comment]
En effet, mais j'avais déjà fait cette remarque pour l'atlas ;)  En fait faire des groupes comme Rhopalos etc est plus compliqué, mieux vaut rester sur le taxref pour éviter de faire des groupes "non génériques" (pour reprendre le vocabulaire de camille ;) ). 

Mais déjà si on pouvais avoir les ordres, on retombe sur quelque chose de clair pour les invertébrés et sans doute la flore.

**Comment by sig-pnrnm** [comment]
>  En fait faire des groupes comme Rhopalos etc est plus compliqué

Certe, ce n'est pas dans TaxRef, mais ce n'est pas si compliqué que ça (en tout cas en France métropolitaine)

Exemple qu'on fait chez nous pour distinguer papillons de jour (rhopalocères) de papillons de nuit (hétérocères) : 
- Papillons de jour : 
`WHERE t.ordre = 'Lepidoptera' AND t.famille NOT IN ('Hesperiidae', 'Lycaenidae', 'Nymphalidae', 'Papilionidae', 'Pieridae', 'Riodinidae');`
- Papillons de nuit : 
`WHERE t.ordre = 'Lepidoptera' AND t.famille NOT IN ('Hesperiidae', 'Lycaenidae', 'Nymphalidae', 'Papilionidae', 'Pieridae', 'Riodinidae');`

Et je me demande d'ailleurs si ce travail n'est pas en cours au Muséum avec la nouvelle table de TaxRef v11 : "`TAXVERNv11`" (référentiel des noms vernaculaires - 53 168 lignes)

Mais bon, les ordres, c'est déjà très bien pour les insectes ! ;)

**Comment by DonovanMaillard** [comment]
Yep, on l'a fait pour notre Atlas (qu'on a arrêté d'alimenter en attendant la V2 puisque tout sera diffusé au niveau régional directement) : http://adherent.flavia-ape.fr/atlas

Mais s'il faut faire ça pour tous les groupes... mais oui, les ordre, j'ai déjà pas mal embêté Camille et Gil avec ça :) En s'y mettant à 2 normands entomos, on va finir par les avoir

**Comment by sig-pnrnm** [comment]
> Mais s'il faut faire ça pour tous les groupes... 

C'est la puissance de l'opensource et du "crowdsourcing" : tu ne peux y parvenir seul (ou alors c'est ch***). Mais en unissant nos forces, c'est pas si dur que ça !

> 
Yep, on l'a fait pour notre Atlas (qu'on a arrêté d'alimenter en attendant la V2 puisque tout sera diffusé au niveau régional directement) : http://adherent.flavia-ape.fr/atlas

Merci pour le lien : c'est toujours sympa de connaitre les réalisations de chacun !
(j'espère que j'aurai prochainement le droit de communiquer le lien de notre Atlas...)

**Comment by sig-pnrnm** [comment]
Sinon, si j'ai bossé sur ce filtre "ergonomique" (tout est relatif), c'est aussi parcequ'il me semblerait pertinant à d'autres endroits de la sphère GeoNature (Atlas).

Par exemple sur les "fiches communes" de GeoNature-Atlas, l'ensemble des espèces de la communes est mélangé. Il serait bien utile de pouvoir filtrer via un simple bouton sur le groupe taxonomique d'intérêt (à l'heure actuelle, on peut trier le tableau par groupe, mais sur plusieurs pages, c'est pas très pratique pour accéder au groupe souhaité).

Voici ce que ça pourrait donner, très rapidement : 

![filtre_taxon_atlas](https://user-images.githubusercontent.com/26482054/43009931-56f6f282-8c3f-11e8-8bac-682d9904b3c7.png)

**Comment by camillemonchicourt** [comment]
Il y a une fonction cachée, dans le champ libre de rechercher tu peux saisir un groupe et ça filtre la liste sur ce groupe :-)

Comme dans la rubrique Photo tu peux rechercher sur un auteur : http://biodiversite.ecrins-parcnational.fr/photos
Autre fonction cachée :-)

**Comment by sig-pnrnm** [comment]
> Il y a une fonction cachée

Sympa !
Ben ya plus qu'à ne plus la cacher ;-)

> Comme dans la rubrique Photo tu peux rechercher sur un auteur

Sympa ça aussi tien !

**Comment by camillemonchicourt** [comment]
C'est parce qu'on a ça affiché dans les bureaux comme leitmotiv ! :-)

![apps](https://img.labnol.org/di/google-apple.jpg)
